com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs

```diff
@@ -479,9 +491,29 @@ public void ToSubGraph()
             if (path.Length == 0)
                 return;
 
+            var nodes = graphView.selection.OfType<IShaderNodeView>().Where(x => !(x.node is PropertyNode || x.node is SubGraphOutputNode)).Select(x => x.node).Where(x => x.allowedInSubGraph).ToArray();
+
+            // Convert To Subgraph could create recursive reference loops if the target path already exists
+            // Let's check for that here
+            if (!string.IsNullOrEmpty(path))
+            {
+                var overwriteGUID = AssetDatabase.AssetPathToGUID(path);
+                if (!string.IsNullOrEmpty(overwriteGUID))
```

Comment:
This path checks for a null GUID while the other one doesn't; I'd love to see these extracted to a single `CheckForRecursiveSubgraph` function.